design/specs/column-extraction.md

```diff
@@ -8,87 +8,66 @@ editor: markdown
+In this case, since we cannot infer the new table name from the selected columns, then the user must manually enter both, new table name and link column name.
 
-    <img width="874" alt="image" src="https://user-images.githubusercontent.com/845767/177550009-30fcae81-2ae9-4539-ad85-5a0d1289790b.png">
+#### Using Rules to Generate Column Names
 
-# Unresolved Design Problems
+In the future we might use a pre-defined list of rules to infer the singular form of the table name, but for now we will require the user to enter both names.
 
-## Extracting a foreign key column to create a mapping table
-If a user had a foreign key referencing a publisher in a publications table and wanted to link to more than one publisher, could they select both the table's primary key and that foreign key and create a new table from both? Should there be a special action that does this without adding a foreign key in place of the columns?
+Applying these rules would require us to choose a specific column from which to generate the final name. If we select the first column, we'll end up with a table called "First Names" and a link column called "First Name."
 
-## Suggesting columns to be extracted
-Offering suggestions for columns that can be extracted into new tables would improve the feature's discoverability and provide users with a clear path to a better data model.
+### Allowing the user to modify the original column selection
 
-## UI for Inspector Panels
-The inspector's UI is still in its early stages and should not be implemented without additional design work.
+Under the `Columns to Extract` section, the user can extend the original column selection by selecting additional columns. The user can also remove columns from the selection.
 
-## Specify the table's name and foreign key column name
-WIP
-## Multiple column actions
-WIP
-## Allow Duplicates
-WIP
-## Selection
-WIP
-### Column vs. Cell Selection
-WIP
-### Row and Column Selection Consistency
-WIP
-## Inspector Modes
-WIP
-### Column
-WIP
-### Cell
-WIP
-### Row
-WIP
-## Keyboard Modifiers
-WIP
-### Touch-only devices
-WIP
-### Visual Cues
-WIP
-## Reordering Columns
-WIP
-## Cell-Only Selection
-WIP
\ No newline at end of file
+This component could be reused for all actions involving column subsets.
+
+Additionally, the user can rename the columns that will be extracted in cases where the column name is not descriptive enough.
+
+## Finalizing the Extract Column Operation
+
+Once the user is happy with the changes, they can click on the `Extract Columns` button. The inspector panel will be toggled and the new link column will be selected in the original table. Under the `Columns` section, the user will see the new table listed as a link.
+
+![image](/assets/design/specs/column-extraction/185143860-f2995fc4-c767-4cce-aaa6-319ef27630ff.png)
+
+Additional details regarding 'Link Properties' will be added in a future spec.
```

Comment:
This doesn't make it clear that `Link Properties` is not meant to be implemented in this spec. Please make that clear.